### PR TITLE
Customization support

### PR DESCRIPTION
**Pull request by arturo-seijas**

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -138,6 +145,7 @@ def _on_pebble_ready(self, event):
 
     def _config_pebble(self, container):
         """Apply pebble changes."""
+        logging.debug("Configuring pebble for container {}".format(container.name))
```
Comment (mthaddon) [inline src/charm.py:148]: This seems like an unrelated change?

File: src/charm.py [hunk 2]
```diff
@@ -222,6 +230,7 @@ def _get_indico_env_config(self):
             "INDICO_SUPPORT_EMAIL": self.config["indico_support_email"],
             "INDICO_PUBLIC_SUPPORT_EMAIL": self.config["indico_public_support_email"],
             "INDICO_NO_REPLY_EMAIL": self.config["indico_no_reply_email"],
+            "CUSTOMIZATION_DEBUG": self.config["customization_debug"],
```
Comment (mthaddon) [inline src/charm.py:233]: Can we alphabetically sort these?

File: src/charm.py [hunk 3]
```diff
@@ -236,9 +245,67 @@ def _on_config_changed(self, event):
...
+            pass
+        return remote_url
+
+    def _download_customization_changes(self):
+        """Clone the remote repository with the customization changes."""
+        current_remote_url = self._get_current_customization_url()
+        if current_remote_url != self.config["customization_sources_url"]:
+            logging.debug(
+                "Removing old contents from directory {}".format(INDICO_CUSTOMIZATION_DIR)
+            )
```
Comment (mthaddon) [inline src/charm.py:275]: See https://juju.is/docs/sdk/logging - "When passing messages to the logger, do not build the strings yourself" - same for all logging messages.

**Review by mthaddon — CHANGES_REQUESTED** [review]
A few comments inline

File: src/charm.py [hunk 4]
```diff
@@ -14,9 +15,12 @@
 from ops.framework import StoredState
 from ops.main import main
 from ops.model import ActiveStatus, MaintenanceStatus, WaitingStatus
+from ops.pebble import ExecError
 
 DATABASE_NAME = "indico"
 PORT = 8080
+CUSTOMIZATION_REFRESH_INTERVAL = 30
```
[the remark this answers is not in the document]
Comment (arturo-seijas) [reply src/charm.py:22]: My bad

File: src/charm.py (hunk 2 above)
Comment (arturo-seijas) [reply src/charm.py:233]: Good point

File: src/charm.py (hunk 3 above)
Comment (arturo-seijas) [reply src/charm.py]: Changed

File: src/charm.py (hunk 1 above)
Comment (arturo-seijas) [reply src/charm.py]: Fixed
